### PR TITLE
Add extension point for tools

### PR DESCRIPTION
**Pull request by gasinvein**
Follow-up for discussion started in flathub/flathub#1417

### CODE REVIEW

**Comment by eszlari** [comment]
Could you add documentation?

**Comment by gasinvein** [comment]
What should I document? Users will see available tools in e.g. Gnome Software, so I guess no install instructions are needed as for SDK extensions.

**Comment by eszlari** [comment]
> Users will see available tools in e.g. Gnome Software, so I guess no install instructions are needed as for SDK extensions.

Not everyone uses Gnome Software.

File: code.sh [hunk 1]
```diff
@@ -13,6 +14,23 @@ if [ ! -f ${FIRST_RUN} ]; then
   touch ${FIRST_RUN}
 fi
 
+for tool_dir in /app/tools/*; do
+  tool_bindir=$tool_dir/bin
+  if [ -d "$tool_bindir" ]; then
+    msg "Adding $tool_bindir to PATH"
+    export PATH=$PATH:$tool_bindir
+  fi
+  tool_pythondir=$tool_dir/lib/python3.7/site-packages
```
Comment (eszlari) [inline code.sh:23]: Could you make this not version specific?

**Comment by eszlari** [comment]
I'm reluctant to merge this, because I'm not sure where this is going.

Take https://github.com/flathub/io.atom.Atom for example:
1. in https://github.com/flathub/io.atom.Atom/commit/ca57fca968af337ce5da2a82a4977704c38fd49d  an extension point was added
2. over a year later, you made it actually usable: https://github.com/flathub/io.atom.Atom/commit/01cafebdb38a810c3cdade13839a4e455fbd6003
3. Where are the extensions?

What extensions do you want to add? Do you think, this will scale?

**Comment by gasinvein** [comment]
I myself would like to add the fish shell, but there are plenty of tools that would be useful the sandbox, while don't fit well to SDK Extension concept, e.g. something like Conan (c/c++ package manager). And anyone could build and install locally whatever they want.
Anyway, I believe extension point without actual extensions is harmless.

File: com.visualstudio.code.yaml [hunk 2]
```diff
@@ -27,6 +27,12 @@ finish-args:
   - --filesystem=xdg-config/kdeglobals:ro
   - --talk-name=com.canonical.AppMenu.Registrar
   - --talk-name=com.canonical.AppMenu.Registrar.*
+add-extensions:
+  com.visualstudio.code.tool:
+    directory: tools
```
Comment (eszlari) [inline com.visualstudio.code.yaml:32]: I would prefer the naming scheme from Atom:
`com.visualstudio.code.Extension`
`directory: extensions`

File: com.visualstudio.code.yaml [hunk 3]
```diff
@@ -53,6 +59,8 @@ modules:
       - ARCH_TRIPLE=$(gcc --print-multiarch) && cp /usr/lib/${ARCH_TRIPLE}/libbfd-*.so
         /app/lib
       - ARCH_TRIPLE=$(gcc --print-multiarch) && ln -s /usr/lib/${ARCH_TRIPLE}/libtinfo.so/app/lib/libtinfo.so.5
+    post-install:
+      - mkdir /app/tools
```
Comment (eszlari) [inline com.visualstudio.code.yaml:63]: The extra `post-install` is not needed.

File: com.visualstudio.code.yaml (hunk 2 above)
Comment (gasinvein) [reply com.visualstudio.code.yaml:32]: IMO `Extension` suggests that it's VS Code extension, but it's not.

Comment (eszlari) [reply com.visualstudio.code.yaml:32]: OK, makes sense.

**Pull request merged**